### PR TITLE
New Rule Implemented for Issue-2445

### LINKED ISSUE #2445: Automatically upgrade AKS cluster node image
### Existing rule

_No response_

### Suggested rule

Use the `SecurityPatch` or `NodeImage` channels with node OS auto-upgrade to maintain secure node images to run cluster components and workloads.

### Pillar

Security

### Additional context

- https://learn.microsoft.com/azure/aks/auto-upgrade-node-image

### PR DESCRIPTION
**Pull request by sharmilamusunuru**
## PR Summary

Added new rule to fix AKS NodeOSAutoUpgrade as part of WAF Security Pillar. 
Fix #2445.

## PR Checklist

- [ X] PR has a meaningful title
- [X ] Summarized changes
- [X ] Change is not breaking
- [ ] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [ ] Unit tests created/ updated
  - [ ] Rule documentation created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

File: docs/en/rules/Azure.AKS.NodeAutoUpgrade.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+<!-- module:config rule AZURE_AKS_CLUSTER_NODE_OS_UPGRADE -->
+
+To configure this rule override the `AZURE_AKS_CLUSTER_NODE_OS_UPGRADE` configuration value to None.
+
+## LINKS
+- [SE:01-Security Baseline](https://learn.microsoft.com/azure/well-architected/security/establish-baseline)
+- [AutoUpgrade NodeImages](https://learn.microsoft.com/azure/aks/auto-upgrade-node-os-image?tabs=azure-cli)
+- [NodeImage Upgrade](https://learn.microsoft.com/azure/aks/node-image-upgrade)
+- [Process Node Updates with Kured](https://learn.microsoft.com/azure/aks/node-updates-kured)
+- [NodeOSUpgrade with GithubActions](https://learn.microsoft.com/azure/aks/node-upgrade-github-actions)
```
Comment (BernieWhite) [inline docs/en/rules/Azure.AKS.NodeAutoUpgrade.md:228]: Additional link.

```suggestion

- [SE:01-Security Baseline](https://learn.microsoft.com/azure/well-architected/security/establish-baseline)
- [AutoUpgrade NodeImages](https://learn.microsoft.com/azure/aks/auto-upgrade-node-os-image?tabs=azure-cli)
- [NodeImage Upgrade](https://learn.microsoft.com/azure/aks/node-image-upgrade)
- [Process Node Updates with Kured](https://learn.microsoft.com/azure/aks/node-updates-kured)
- [NodeOSUpgrade with GithubActions](https://learn.microsoft.com/azure/aks/node-upgrade-github-actions)
- [Azure deployment reference](https://learn.microsoft.com/azure/templates/microsoft.containerservice/managedclusters)
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Thanks @sharmilamusunuru. Great work!

See below for a few suggestions (you can click commit suggestion to automatically apply these changes), otherwise all good to merge.

**Comment by sharmilamusunuru** [comment]
Thanks @BernieWhite for review of PR, suggestions are added and PR is ready to merge

**Review by BernieWhite — APPROVED** [review]
Thanks @sharmilamusunuru. All good to merge.

**Pull request merged**